insights/client/apps/compliance/__init__.py

```diff
@@ -84,35 +85,37 @@ def os_release(self):
     def profile_files(self):
         return glob("{0}*rhel{1}*.xml".format(POLICY_FILE_LOCATION, self.os_release()))
 
-    def find_scap_policy(self, profile_ref_id):
-        rc, grep = call('grep ' + profile_ref_id + ' ' + ' '.join(self.profile_files()), keep_rc=True)
+    def find_scap_policy(self, parent_profile_ref_id):
+        rc, grep = call('grep ' + parent_profile_ref_id + ' ' + ' '.join(self.profile_files()), keep_rc=True)
         if rc:
-            logger.error('XML profile file not found matching ref_id {0}\n{1}\n'.format(profile_ref_id, grep))
+            logger.error('XML profile file not found matching ref_id {0}\n{1}\n'.format(parent_profile_ref_id, grep))
             exit(constants.sig_kill_bad)
         filenames = findall('/usr/share/xml/scap/.+xml', grep)
         if not filenames:
-            logger.error('No XML profile files found matching ref_id {0}\n{1}\n'.format(profile_ref_id, ' '.join(filenames)))
+            logger.error('No XML profile files found matching ref_id {0}\n{1}\n'.format(parent_profile_ref_id, ' '.join(filenames)))
             exit(constants.sig_kill_bad)
         return filenames[0]
 
-    def build_oscap_command(self, profile_ref_id, policy_xml, output_path, tailoring_file_path):
-        command = 'oscap xccdf eval --profile ' + profile_ref_id
+    def build_oscap_command(self, parent_profile_ref_id, policy_xml, output_path, tailoring_file_path):
+        command = 'oscap xccdf eval --profile ' + parent_profile_ref_id
         if tailoring_file_path:
             command += ' --tailoring-file ' + tailoring_file_path
         command += ' --results ' + output_path + ' ' + policy_xml
         return command
 
-    def run_scan(self, profile_ref_id, policy_xml, output_path, tailoring_file_path=None):
+    def run_scan(self, profile_ref_id, parent_profile_ref_id, policy_xml, output_path, tailoring_file_path=None):
         logger.info('Running scan for {0}... this may take a while'.format(profile_ref_id))
         env = os.environ.copy()
         env.update({'TZ': 'UTC'})
-        oscap_command = self.build_oscap_command(profile_ref_id, policy_xml, output_path, tailoring_file_path)
+        oscap_command = self.build_oscap_command(parent_profile_ref_id, policy_xml, output_path, tailoring_file_path)
         rc, oscap = call(oscap_command, keep_rc=True, env=env)
         if rc and rc != NONCOMPLIANT_STATUS:
             logger.error('Scan failed')
             logger.error(oscap)
             exit(constants.sig_kill_bad)
         else:
+            if (profile_ref_id != parent_profile_ref_id):
+                self._replace_profile_ref_id(output_path, parent_profile_ref_id, profile_ref_id)
```

Comment:
Before adding to the archive, substitute ref_id for parent_profile_ref_id if they're different.